topdown/eval.go

```diff
@@ -2268,7 +2271,28 @@ func (e evalVirtual) eval(iter unifyIterator) error {
 			empty:     ast.SetTerm(),
 		}
 		return eval.eval(iter)
-	case ast.PartialObjectDoc:
+	case ast.SingleValue:
```

Comment:
This is the adaptation -- we're checking if any of the rules has a non-ground key in its last position, and if so, use the partialobjectdoc code path.